google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -3343,6 +3578,27 @@ private void createSessions(final int sessionCount, boolean distributeOverChanne
     }
   }
 
+  class MultiplexedSessionConsumer implements SessionConsumer {
+
+    @Override
+    public void onSessionReady(SessionImpl sessionImpl) {
+      final MultiplexedSession session = new MultiplexedSession(sessionImpl);
+      synchronized (lock) {
+        multiplexedSession.set(session);
+        numMultiplexedSessionsBeingCreated = false;
+        while (!multiplexedSessionWaiters.isEmpty()) {
+          multiplexedSessionWaiters.poll().put(multiplexedSession.get());
+        }
+      }
+    }
+
+    @Override
+    public void onSessionCreateFailure(Throwable t, int createFailureForSessionCount) {
+      numMultiplexedSessionsBeingCreated = false;
```

Comment:
This should only be updated while holding the lock

Reply:
Done.